popup_filter_build.js

```
@@ -0,0 +1,54 @@
+const { promises: fs } = require('fs');
+
+/**
+ * Path to the filter file
+ * @type {string}
+ */
+const filterPath = process.argv[2];
+
+/**
+ * Function to convert modifiers in a rule
+ * @param {string} rule - The rule to modify
+ * @param {string} modifier - The new modifier
+ * @param {string} [modifierValue] - The value for the modifier
+ * @returns {string} - Modified rule
+ */
+const convertModifier = (rule, modifier, modifierValue) => {
+    // match all letters after the $ sign until the end of the line (modifier)
+    const regex = /\$[a-z,]{0,15}$/;
+    return rule.replace(regex, `$${modifier}${modifierValue ? `=${modifierValue}` : ''}`);
+};
+
+/**
+ * Function to get file content as an array of lines
+ * @param {string} path - Path to the file
+ * @returns {Promise<string[]>} - Promise resolving to an array of lines from the file
+ */
+const getFileContent = async (path) => {
+    try {
+        const content = await fs.readFile(path, 'utf-8');
+        return content.split('\n');
+    } catch (error) {
+        console.error('Error reading file:', error);
+        return [];
```

Do not ignore errors, throw them